src/main/java/org/spongepowered/api/resource/Resource.java

```
@@ -0,0 +1,178 @@
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package org.spongepowered.api.resource;
+
+import com.google.common.io.ByteStreams;
+import com.google.common.io.CharStreams;
+import com.google.common.io.MoreFiles;
+import org.spongepowered.api.data.DataView;
+import org.spongepowered.api.data.persistence.DataFormat;
+import org.spongepowered.api.data.persistence.InvalidDataFormatException;
+
+import java.io.BufferedReader;
+import java.io.Closeable;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.io.OutputStream;
+import java.nio.charset.Charset;
+import java.nio.file.OpenOption;
+import java.nio.file.Path;
+import java.util.Optional;
+import java.util.stream.Stream;
+import javax.annotation.WillNotClose;
+
+/**
+ * A resource can represent any kind of loaded data. It can be a file on the
+ * filesystem, a network location, or even generated at runtime. Use
+ * {@link #getInputStream()} to load the data held by a resource.
+ *
+ * <p>Resource extends Closeable, so don't forget to close your streams either
+ * with a try-with-resources or a close() inside a try-finally block. e.g.</p>
+ * <pre>
+ *     try (Resource res = resourceManager.getResource(path)) {
+ *         logger.info(res.readString(DefaultCharsets.UTF_8));
+ *     } catch (IOException e) {
+ *         e.printStackTrace();
```

We should recommend users to log the error or throw it instead of `printStackTrace`